src/nrncvode/netcvode.cpp

```diff
@@ -1674,20 +1703,39 @@ bool NetCvode::init_global() {
                         int icell = cellnum[ml->nodelist[j]->v_node_index];
                         if (cvml[icell]->index != i) {
                             cvml[icell] = cvml[icell]->next;
-                            assert(cvml[icell] && cvml[icell]->index);
+                            assert(cvml[icell] && cvml[icell]->index == i);
                         }
-                        cml = cvml[icell];
-                        auto& newml = cml->ml.emplace_back(cml->index /* mechanism type */);
-                        newml.nodecount = 1;
-                        newml.nodelist = new Node*[1];
-                        newml.nodelist[0] = ml->nodelist[j];
-                        newml.nodeindices = new int[1]{ml->nodeindices[j]};
-                        newml.prop = new Prop* [1] { ml->prop[j] };
-                        if (!mf.hoc_mech) {
-                            newml.set_storage_offset(ml->get_storage_offset() + j);
-                            newml.pdata = new Datum* [1] { ml->pdata[j] };
+                        auto& cml = cvml[icell];
+                        int kk = j;
+                        for (auto& newml: cml->ml) {
```

Comment:
Any way to extract this whole loop into its own separate (probably `inline` so not to lose perf) method? The amount of indentation is becoming a bit unwieldy.

Reply:
I've incorporated the improvements mentioned above.